Add documentation with using StackHPC helm chart instead

DavidFair opened the pull request:
Replaces the old way of deploying with a new StackHPC deployment method that simplifies things from a user POV

DavidFair committed:
Add documentation with using StackHPC helm chart instead

Replaces the old way of deploying with a new StackHPC deployment method
that simplifies things from a user POV

meoflynn commented on source/ClusterAPI/Setup.rst:
```diff
@@ -55,192 +45,132 @@ cloud Ubuntu image, not the stripped down CAPI image designed for nodes.
+        auth_type: "v3applicationcredential"
 
-To the floating IP pre-allocated in the project:
+- Add the UUID of the project to create the cluster in. This can be found `here <https://openstack.stfc.ac.uk/identity/>`_.
```
Where does this project uuid need to be added?

meoflynn commented on source/ClusterAPI/Setup.rst:
```diff
@@ -12,16 +12,6 @@ Setup a new cluster using Cluster API
 Deployment Considerations
 =========================
 
-Deployment Account
-------------------
-
-The deployment is tied to the account which created it. This includes their
-credentials being encoded as an opaque secret within the cluster for the cluster API openstack component.
-
-For production workloads, and long-lived clusters with multiple accounts a service account it recommended.
-This means the cluster isn't tied to the lifetime of a single user's account or password. A service
-account can be requested through our :ref:`contact_details`.
-
```
I think we should keep the original CAPI docs alongside the new docs for now - maybe keep the original docs in a separate page?

DavidFair replied:
I would argue not, it's in source control and I'm not keen to support this method at all going forwards

I'm happy to talk about it offline though if you feel strongly :slightly_smiling_face:

meoflynn commented on source/ClusterAPI/Setup.rst:
```diff
@@ -33,7 +23,7 @@ to quickly gain access and perform recovery or upgrades as required.
 ----------------
 
-A `clouds.yaml` file with the user's password is also required. This file should be removed or restricted
+A `clouds.yaml` file with the application credentials is also required. This file should be removed or restricted
```
Should we also mention for application credentials when they are created they should **never** be unrestricted application credentials?

meoflynn commented on source/ClusterAPI/Setup.rst:
```diff
@@ -55,192 +45,132 @@ cloud Ubuntu image, not the stripped down CAPI image designed for nodes.
-    kubectl apply -f $CLUSTER_NAME.yaml
+Configuring the cluster
+-----------------------
 
```
Maybe we could add a summary of what each YAML file is used for by the Helm chart here?

DavidFair committed:
Documentation improvements from feedback

Improve docs with some additional details about CAPI based on PR
feedback

DavidFair committed:
Feedback from internal deployment

Applies feedback gathered from running through these docs internally
within the dev ops team

meoflynn committed:
Fix formatting

DavidFair committed:
Switch to minikube from KinD

This is easier to setup, and it's more likely for users know it

DavidFair committed:
Add missing install step for CRDs

This is required for the subsequent install step

meoflynn committed:
Merge pull request #98 from stfc/switch_to_minikube

CAPI: Switch to minikube
